### PR TITLE
Automate BW-items generation for build01

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-721

/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  local config
+  config=$2
+  local registry
+  registry=$3
+  local token
+  token=$(oc --kubeconfig "${config}" sa get-token -n ci build01)
+  local auth_file
+  auth_file="${WORKDIR}/auth_file.${cluster}.config"
+  podman login -u unused -p "${token}" --authfile "${auth_file}" "${registry}"
+  jq --arg reg "${registry}" -r '.auths[$reg].auth' "${auth_file}" > "${WORKDIR}/build01_${cluster}_reg_auth_value.txt"
```
Comment (stevekuznetsov) [inline clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Try

```
oc get secret $( oc get secret -n ci -o "jsonpath={.items[?(@.metadata.annotations.kubernetes\.io/service-account\.name==\"${serviceaccount}\")].metadata.name}" ) -o "jsonpath={.items[?(@.type==\"kubernetes.io/dockercfg\")].data.\.dockercfg}" | base64 --decode
```

Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Done.

**Comment by hongkailiu** [comment]
```bash
$ ll /tmp/tmp.vUEGJORret/
total 32
-rw-rw-r--. 1 liu liu 1125 Jan  3 16:10 build01_build01_reg_auth_value.txt
-rw-------. 1 liu liu 1472 Jan  3 16:10 sa.ca-cert-issuer.build01.config
-rw-------. 1 liu liu 1706 Jan  3 16:10 sa.ci-operator.build01.config
-rw-------. 1 liu liu 1412 Jan  3 16:10 sa.deck.build01.config
-rw-------. 1 liu liu 1412 Jan  3 16:10 sa.hook.build01.config
-rw-------. 1 liu liu 1810 Jan  3 16:10 sa.kubeconfig
-rw-------. 1 liu liu 1418 Jan  3 16:10 sa.plank.build01.config
-rw-------. 1 liu liu 1424 Jan  3 16:10 sa.sinker.build01.config
```

Screen Shot 2020-01-03 at 4 32 47 PM


`build01_build01_reg_auth_value.txt` is from api.ci cluster, not relevant to recreation of `build01`.

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+declare -a SAArray=( "config-updater" "deck" "plank" "sinker" "hook" "ci-operator" "ca-cert-issuer" )
+ 
+# Iterate the string array using for loop
+for name in ${SAArray[@]}; do
+   generate_kubeconfig ${name}
+done
+
+# config-updater is the first one we set up ... naming is not with a convention
```
Comment (stevekuznetsov) [inline clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Can we fix this first?

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+### 3. jq is installed
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+WORKDIR="$(mktemp -d)"
+readonly WORKDIR
+
+### change me
```
Comment (stevekuznetsov) [inline clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Remove this from the production script

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Yes.
/hold

Blocked by the fix.
Will be another PRs.

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh (hunk 3 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Done.

File: clusters/build-clusters/01_cluster/hack/generate-bw-items.sh (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/hack/generate-bw-items.sh]: Done.
https://github.com/openshift/release/pull/6560

**Pull request merged**